services/server/server.go

```diff
@@ -116,81 +117,34 @@ func (s *Server) UpdateSettingsFromEnv(env []string) error {
 			return err
 		}
 
-		for _, e := range env {
-			p := strings.SplitN(e, "=", 2)
-			if len(p) != 2 {
-				s.l.Warnf("Failed to parse environment variable %q.", e)
-				continue
+		envSettings, errs, warns := validators.EnvVarValidator(env)
```

Comment:
I think we should move that call out of transaction. Parse variables, set `s` fields, then get settings and update them with `s` fields.

Reply:
Refactored.